x-pack/plugins/maps/public/classes/sources/es_agg_source/es_agg_source.ts

```diff
@@ -68,11 +72,16 @@ export class AbstractESAggSource extends AbstractESSource {
     return FIELD_ORIGIN.SOURCE;
   }
 
-  getMetricFields(): IESAggField[] {
+  getMetricFields(canReadFromGeoJson = true): IESAggField[] {
```

Comment:
Why is canReadFromGeoJson passed in as a parameter? Can canReadFromGeoJson ever be different from constructor canReadFromGeoJson? Why not just use canReadFromGeoJson from the constructor here?